corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -0,0 +1,336 @@
+            + "--maxWriteSize=<maxWriteSize> Max write size smaller than 2GB\n"
+            + "--bulkReadSize=<bulkReadSize> Read size for chain replication\n"
+            + "--trim=<trim> Should trim be performed in this run\n"
+            + "--isUpgrade=<isUpgrade> Is this called during upgrade\n"
+            + "--trimTokenFile=<trimTokenFilePath> file to store the trim tokens during upgrade"
+            + "--upgradeDescriptorTable=<upgradeDescriptorTable> Repopulate descriptor table?\n"
+            + "--tlsEnabled=<tls_enabled>";
+
+    public CorfuStoreCompactorMain() {
+        CorfuRuntimeHelper corfuRuntimeHelper;
+        CorfuRuntimeHelper cpRuntimeHelper;
+        if (tlsEnabled) {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+            cpRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+
+        } else {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+            cpRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+        }
+
+        corfuRuntime = corfuRuntimeHelper.getRuntime();
+        cpRuntime = cpRuntimeHelper.getRuntime();
+        corfuStore = new CorfuStore(corfuRuntime);
+        distributedCompactor = new DistributedCompactor(corfuRuntime, cpRuntime, persistedCacheRoot);
+
+        this.openCompactionTables();
+    }
+
+    private void upgrade() {
+        retryCheckpointing = 10;
+        if (upgradeDescriptorTable) {
+            syncProtobufDescriptorTable();
+        }
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            txn.putRecord(checkpointTable, DistributedCompactor.UPGRADE_KEY, TokenMsg.getDefaultInstance(),
+                    null);
+            txn.commit();
+        } catch (Exception e) {
+            log.warn("Unable to write UpgradeKey to checkpoint table, ", e);
+        }
+    }
+
+    /**
+     * Entry point to invoke Client checkpointing by the CorfuServer
+     *
+     * @param args command line argument strings
+     */
+    @SuppressWarnings("UncommentedMain")
+    public static void main(String[] args) throws Exception {
+        getCompactorArgs(args);
+
+        if (maxWriteSize == -1) {
+            if (persistedCacheRoot == null) {
+                // in-memory compaction
+                maxWriteSize = DEFAULT_CP_MAX_WRITE_SIZE;
+            } else {
+                // disk-backed non-config compaction
+                maxWriteSize = NON_CONFIG_DEFAULT_CP_MAX_WRITE_SIZE;
+            }
+        }
+
+        if (port == 9040) {
```

Comment:
why magic number? And this thing: config/non config shouldn't be in the source code.
We need to find a way how to avoid using this piece of code

Reply:
yes agreed!